plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -430,7 +430,9 @@ class bp_connection_manager {
             }
          } else {
             auto r = idx.equal_range(peer.producer_name);
-            if (std::distance(r.first, r.second) >= max_bp_gossip_peers_per_producer) {
+            // The static_cast is to prevent the sign-compare compile
+            // warning. It is a safe cast.
+            if (std::distance(r.first, r.second) >= static_cast<std::ptrdiff_t>(max_bp_gossip_peers_per_producer)) {
```

Comment:
would c++20's `std::cmp_greater_equal()` be a better choice (as it eliminates the cast and might even be more pedantically correct mathematically)? I haven't really used these std::cmp*() funcs but initially seems exactly like what we might want for these sort of situations